### DISCUSSION TITLE
Software Support | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1121] [Q&A]
# Software Support | Seeed Studio Wiki

Getting Started with Grove Vision AI V2 and software platform

https://wiki.seeedstudio.com/grove_vision_ai_v2_software_support/

**Comment by Love4yzp** [comment]
There is an example using Arduino running on RP2040 to work with Grove Vision V2.
[.ino file](https://github.com/Seeed-Solution/SenseCAP_Indicator_ESP32/blob/feat-vision-v2/examples/vision_v2_display/RP2040/RP2040.ino)

Anyone wants to see the result?

**Comment by neoscell** [comment]
For demo 2, need to add AI.begin() after establishing serial:

void setup()
{
    // put your setup code here, to run once:
    Wire.begin();
    Serial.begin(115200);
    AI.begin();
    while (!Serial)
        delay(1000);

    Serial.println("Proxy start");
}

**Comment by Gandhuttt** [comment]
anyone else having issue while invoking the grove vision when connecting it to the xiao esp32c3 board? its fine when i invoke grove vision as standalone without the xiao board in https://seeed-studio.github.io/SenseCraft-Web-Toolkit/#/setup/process or https://sensecraft.seeed.cc/ai/device/local/36, but when i invoke it while its connected to the xiao board the camera looks to be frozen, could be a firmware issue?

**Reply by ReKST-117** [reply]
Hi, I am assuming the SenseCraftAI platform only support Grove Vision AI Camera or XIAO esp32s3 Sense. I tried both of them and worked. So if you want to use XIAO+Grove Vision. You need to use Arduino IDE following the tutorial: https://wiki.seeedstudio.com/grove_vision_ai_v2_software_support/#-getting-started-with-seeed-studio-xiao-
Also, I don't think esp32c3 is compatible with the platform as when you connect the device, the device selection list explicitly stated that XIAO esp32s3; Grove Vision AI V2 and SenseCAP A1102. XIAO esp32c3 is not listed here. 
<img width="192" alt="7 4 vi5" src="https://github.com/user-attachments/assets/0b11e1c1-dd44-4641-abbf-bff148466385" />

**Comment by marcusobrien** [comment]
I have the XIAO ESP32-S3 plugged onto the AI Vision V2 board - BOTH the ESP32-S3 and AI Vision v2 are powered through USB - the USB to the ESP32-S3 is connected to a PC to provide UART comms so I can see the log from the firmware running on the ESP32.

I first loaded the Gesture Model on to the AI board using the sensecraft web portal - https://sensecraft.seeed.cc/ai/device/local/36?id=60111&uniform_type=36&name=Gesture%20Detection&adapteds=11&adapteds=12&adapteds=14&task=1 

I then used Arduino IDE 2 to load the demo 1 example above, and also added some UART output so I can make sure my Serial Monitor is connected properly to the ESP32-S3, so my start function looks like this

void setup()
{
    Serial.begin(9600);
    while (!Serial) delay(10);
    AI.begin();
    delay(2000); 
    Serial.println("ESP32-S3 setup Serial working.");
}

When I upload and run this code then connect to the Serial Monitor inside the Arduino IDE v2 (at baud 9600) , I can see this output
ESP32-S3 setup Serial working.

BUT the code inside loop() from the demo does NOT work. I call AI.invoke and I always get 3, if I call AI.available I get 0, I added more serial for debug 

void loop()
{
    int invoke = AI.invoke();
    Serial.printf("LoopStart invoke is %d\n",invoke);

Output is 
LoopStart invoke is 3

If I change it to always call both AI.invoke() and AI.available() to see the state of both

void loop()
{
    int invoke = AI.invoke();
    Serial.printf("LoopStart invoke is %d\n",invoke);

    int avail = AI.available();
    Serial.printf("LoopStart available:%d \n",avail);
         
The above Serial is 
LoopStart invoke is 3
LoopStart available:0

Obviously I am putting signs in front of the camera, and everything works when using the web portal - the green LED is on the AI Vision 2, and both that and XIAO ESP32-S3 are powered on - so how can i tell if the model is loaded and running from the XIAO ? How can I get this working ?

**Reply by marcusobrien** [reply]
BTW - when I use a XIAO ESP32-C3 (the C3) not the XIAO ESP32-S3 (the S3) ir works perfectly. So it appears its either the ESP32 S3 I have (I will try another one), OR the demo above and code only works with the XIAO ESP32 C3 only

**Reply by marcusobrien** [reply]
I have fixed this - looks like a loose pin onm the XIAO ESP32-S3, so now both the C3 and the S3 work with the AI Vision 2.

**Reply by nfs0619** [reply]
Glad to hear you’ve resolved the issue! If you run into any other problems, feel free to contact our technical support at **[techsupport@seeed.io](mailto:techsupport@seeed.io)**.

**Comment by njuhang** [comment]
Which app should I use when I use SenseCraft AI to help with live streaming showing?

**Reply by nfs0619** [reply]
Thanks for your question. When using SenseCraft AI, the live camera preview will be shown directly in the browser after the device is connected, so no additional app is required.

**Reply by njuhang** [reply]
Sorry, I wasn't clear enough. 
I mean which app should I use in this repo: https://github.com/HimaxWiseEyePlus/Seeed_Grove_Vision_AI_Module_V2? Is it the default app names allon_sensor_tflm?
